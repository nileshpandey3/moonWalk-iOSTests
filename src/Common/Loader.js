import React, { Component } from "react";
import styled from "styled-components";
import { Animated } from "react-native";

const FullWidthWrapper = styled.View`
  align-items: center;
  margin: 10px;
`;

const Wrapper = styled(Animated.View)`
  align-items: center;
  width: 46px;
`;

const Center = styled.View`
  height: 24px;
  width: 24px;
  border-radius: 12px;
  background: white;
`;

const OrbitLine = styled.View`
  height: 46px;
  width: 46px;
  border-radius: 23px;
  align-items: center;
  justify-content: center;
  border-width: 2px;
  border-color: white;
`;

const Moon = styled.View`
  height: 14px;
  width: 14px;
  border-radius: 7px;
  background: white;
  position: absolute;
  left: 34px;
  top: 0;
`;

export default class extends Component {
  state = {
    moonAnim: new Animated.Value(0),
    appearAnim: new Animated.Value(0)
  };

  componentDidMount() {
    Animated.loop(
      Animated.timing(this.state.moonAnim, { toValue: 1, duration: 1000 })
    ).start();
    Animated.timing(this.state.appearAnim, {
      toValue: 1,
      duration: 400
    }).start();
  }

  render() {
    const moonRotation = this.state.moonAnim.interpolate({
      inputRange: [0, 1],
      outputRange: ["0deg", "360deg"]
    });

    return (
      <FullWidthWrapper>
        <Wrapper
          style={{
            transform: [{ rotate: moonRotation }],
            opacity: this.state.appearAnim
          }}
        >
          <OrbitLine>
            <Center />
          </OrbitLine>
          <Moon />
        </Wrapper>
      </FullWidthWrapper>
    );
  }
}
