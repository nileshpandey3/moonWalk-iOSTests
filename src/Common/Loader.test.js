import React from "react";
import Loader from "./Loader";
import renderer from "react-test-renderer";

jest.useFakeTimers();

describe("Loader", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Loader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
