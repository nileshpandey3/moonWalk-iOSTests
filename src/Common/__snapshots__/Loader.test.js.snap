// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "opacity": 0,
        "transform": Array [
          Object {
            "rotate": "0deg",
          },
        ],
        "width": 46,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "white",
            "borderRadius": 23,
            "borderWidth": 2,
            "height": 46,
            "justifyContent": "center",
            "width": 46,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 12,
              "height": 24,
              "width": 24,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderRadius": 7,
            "height": 14,
            "left": 34,
            "position": "absolute",
            "top": 0,
            "width": 14,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
